File: MainService/src/main/java/com/reins/bookstore/service/CartService.java
package com.reins.bookstore.service;

import com.reins.bookstore.entity.Book;
import com.reins.bookstore.entity.Cart;
import com.reins.bookstore.entity.CartDto;

import java.util.List;


public interface CartService {
    List<Cart> getCartsByUser(Integer userId);
    void addCart(CartDto cartDto);

    void removeCartsByUser(Integer userId);
}

File: src/services/cartService.js
import { postRequest_v2 } from "../utils/ajax";
export const getCartsByUser = ({ search }, callback) => {
    const data = { userId: search };
    // const url = `${config.apiUrl}/getOrdersByUser`;
    const url = `http://localhost:8080/getCartsByUser`;
    // console.log("getOrdersByUser");
    // console.log(data);
    postRequest_v2(url, data, callback);
    // console.log("service");
    // console.log(data);
  };


  export const addCart = (cartDto, callback) => {
    const url = `http://localhost:8080/addCart`;
    console.log("addCart");
  
    postRequest_v2(url, cartDto, callback);
  }


  export const removeCartsByUser = (userId, callback) => {
    const data = { userId: userId };
    // const url = `${config.apiUrl}/getOrdersByUser`;
    const url = `http://localhost:8080/removeCartsByUser`;
    // console.log("getOrdersByUser");
    // console.log(data);
    postRequest_v2(url, data, callback);
    // console.log("service");
    // console.log(data);
  };
  